webapp/components/sidebar.jsx

```diff
@@ -116,14 +116,19 @@ export default class Sidebar extends React.Component {
         const currentChannelId = ChannelStore.getCurrentId();
         const tutorialStep = PreferenceStore.getInt(Preferences.TUTORIAL_STEP, UserStore.getCurrentId(), 999);
 
-        const allChannels = ChannelStore.getAll().map((channel) => Object.assign({}, channel));
-        const channelList = ChannelUtils.buildDisplayableChannelList(allChannels);
+        const channels = ChannelStore.getAll();
+        let displayableChannels = {};
+        if (channels !== this.oldChannels) {
+            const channelsArray = channels.map((channel) => Object.assign({}, channel));
+            displayableChannels = ChannelUtils.buildDisplayableChannelList(channelsArray);
+        }
+        this.oldChannels = channels;
 
         return {
             activeId: currentChannelId,
             members,
             teamMembers,
-            ...channelList,
+            ...displayableChannels,
```

Comment:
Nope, `this.state.displayableChannels` doesn't exist. If `displayableChannels` is {} then nothing changes, if it is {favoriteChannels, publicChannels, etc.} then it will update `this.state.favoriteChannels`, `this.state.publicChannels`, etc.